dask/dataframe/io.py

```diff
@@ -724,10 +736,59 @@ def one_path_one_key(path, key, start, stop, columns, chunksize, lock):
     if (start != 0 or stop is not None) and len(keys) > 1:
         raise NotImplementedError(read_hdf_error_msg)
     from .multi import concat
-    return concat([one_path_one_key(path, k, start, s, columns, chunksize, lock)
+    return concat([one_path_one_key(path, k, start, s, columns, chunksize,
+                                    lock, division_column)
                    for k, s in zip(keys, stops)])
 
 
+def _find_divisions(division_column, start, stop, chunksize, path, key, lock):
+    divisions = list()
+    starts = [start]
+    stops = list()
+    row = start
+    if isinstance(division_column, string_types):
+        _div_col = lambda x: x[division_column]
+    elif division_column:
+        _div_col = lambda x: pd.Series(x.index)
+    stop = _get_last_row(path, key, lock, stop)
+    data = _pd_read_hdf(path, key, lock,
+                        {'start': row, 'stop': row + 1})
+    divisions.append(_div_col(data).iloc[0])
+    for row in range(start + chunksize, stop, chunksize):
+        data = _pd_read_hdf(path, key, lock,
+                            {'start': row - 1, 'stop': row + 1})
```

Comment:
Does pandas let us access the values directly from the HDF file?  How expensive is it to make a new pandas dataframe every time like this?  There will be many cases where we need to scan thousands of elements before we find a change.

Also, feel free to not use the internal functions here like `_pd_read_hdf` , they're a bit janky.  I would use the standard Pandas API directly.  The internal functions solve annoying problems that come up when computing dask graphs.